NetUtilities/System/IO/MemoryStreamExtensions.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+
+namespace System.IO
+{
+    /// <summary>
+    /// A extension class for extensions to <see cref="MemoryStream"/>.
+    /// </summary>
+    public static class MemoryStreamExtensions
+    {
+        /// <summary>
+        /// Clears the data in the <see cref="MemoryStream"/>.
+        /// </summary>
+        /// <param name="ms">
+        /// The input <see cref="MemoryStream"/>.
+        /// </param>
+        public static void Clear(this MemoryStream ms, int capacity)
+        {
+            if (ms == null)
+            {
+                throw new ArgumentNullException(nameof(ms));
+            }
+
+            // false if they are not equal or if capacity is 0.
+            bool changeCapacity = ms.GetBuffer().Length != capacity || capacity == 0;
+            Array.Clear(ms.GetBuffer(), 0, ms.GetBuffer().Length);
```

Comment:
I changed this as not sure if Array.Clear would only change only the local byte array, or the array owned by the MemoryStream when I did it the other way 😅. This way makes it more explicit but I feel I could save the length to a local variable too,